woodwork/logical_types.py

```diff
@@ -356,6 +364,11 @@ def transform(self, series, null_invalid_values=False):
                     )
 
         series = self._remove_timezone(series)
+        if self.datetime_format is not None and "%y" in self.datetime_format:
```

Comment:
Love how much simpler this is. I think we should keep an eye and just make sure we're getting date formats the right way once Jeff's stuff goes in, but hopefully this will integrate nicely.